src/PrestaShopBundle/Translation/Extractor/LegacyFileExtractor.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation\Extractor;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+
+final class LegacyFileExtractor implements LegacyFileExtractorInterface
+{
+    /**
+     * @param string $path
+     * @param string $locale
+     *
+     * @return MessageCatalogueInterface
+     */
+    public function extract($path, $locale)
+    {
+        $_MODULE = [];
+        $locale = $this->convertToCatalogueLocale($locale);
+        $filepath = $path . "$locale.php";
+
+        if (!file_exists($filepath)) {
+            throw new \Exception('There is no translation file available.');
+        }
+
+        include_once $filepath;
```

Comment:
this file contains a global array named $_MODULE... I'd like to encapsulate it into a local variable instead: any idea?

I wanted to use the PHP parser, but the gains aren't worth the efforts :thinking:

Reply:
Also, what if the file is invalid? this isn't supposed to happens: so it will happens ...

Reply:
@mickaelandrieu it's not that hard to make this file invalid, i remember number of time when people edited translations directly in some sh@#@!# editor which break UTF encoding, PrestaShop couldn't handle this file anymore

Reply:
@mickaelandrieu also, i'm wondering, what if new phrase is added to code? this phrase is not automatically visible in /modules/name/translations/iso.php AFAIK

Reply:
probably this is why PrestaShop has to scan all .php files in legacy translation mechanism

Reply:
> this phrase is not automatically visible in /modules/name/translations/iso.php AFAIK

Be patient, I'm adding 1 commit really soon with a LegacyModule Extractor :+1: : let me time to write my unit tests boss!